hal/ia32/cpu.c

```diff
@@ -392,22 +418,36 @@ void *_cpu_initCore(void)
 {
 	cpu.ncpus++;
 
-	u32 a = *(u32 *)0xfee000f0;
-	*(u32 *)0xfee000f0 = (a | 0x100);
+	const unsigned int id = hal_cpuGetID();
+	/* 0xfee000f0 - Local APIC, Spurious Interrupt Vector Register */
+	volatile u32 *p = (void *)0xfee000f0;
+	*p = (*p | 0x100);
 
-	hal_memset(&cpu.tss[hal_cpuGetID()], 0, sizeof(tss_t));
+	hal_memset(&cpu.tss[id], 0, sizeof(tss_t));
 
-	_cpu_gdtInsert(TLS_DESC_IDX + cpu.ncpus, (u32)&cpu.tss[hal_cpuGetID()], sizeof(tss_t), DESCR_TSS);
+	_cpu_gdtInsert(hal_cpuGetTssIndex(), (u32)&cpu.tss[id], sizeof(tss_t), DESCR_TSS);
+	_cpu_gdtInsert(hal_cpuGetTlsIndex(), 0x00000000, VADDR_KERNEL, DESCR_TLS);
 
-	cpu.tss[hal_cpuGetID()].ss0 = SEL_KDATA;
-	cpu.tss[hal_cpuGetID()].esp0 = (u32)&cpu.stacks[hal_cpuGetID()][511];
+	cpu.tss[id].ss0 = SEL_KDATA;
+	cpu.tss[id].esp0 = (u32)&cpu.stacks[id][511];
+
+	/* Init FPU - set flags:
+	   MP - FWAIT instruction does not ignore TS flag
+	   TS - The first use of FPU generates device-not-available exception (#NM)
+	   NE - FPU exceptions are handled internally*/
+	__asm__ volatile(
```

Comment:
Please adapt style of this inline asm to the style found below (apart form `volatile(`, should be `volatile (`):
```
	__asm__ volatile (
		"fninit;"
		"movl %%cr0, %%eax;"
		"orl $0x2A, %%eax;"
		"mov %%eax, %%cr0;"
	::: "eax");
```
or, both to style I personally prefer:
```
	__asm__ volatile (
		"fninit;"
		"movl %%cr0, %%eax;"
		"orl $0x2A, %%eax;"
		"mov %%eax, %%cr0;"
	:
	:
	: "eax");
```

Reply:
Side note `\n\t` (line break, tab) instead of `;` and instead `\` should be used. Reading, analysing assembly listening generated by `gcc -S` where either`;` or `\` is used as line delimiter is painful because all instructions are placed in single line) .   I know it is ugly, but the only correct way (see [gcc:inlineassembly](https://gcc.gnu.org/onlinedocs/gcc/extensions-to-the-c-language-family/how-to-use-inline-assembly-language-in-c-code.html#parameters)).
Also `;` in assembly is reserved for comments (some assemblers use it as start of comment line)

How unreadable `gcc -S` output looks (marked with red rectangle):
![2023-05-29-191211_944x1155_scrot](https://github.com/phoenix-rtos/phoenix-rtos-kernel/assets/141153/29a02d23-01db-4672-97cd-adeace458ddb)

Reply:
@gerard5 does this style cause the same problem: https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/c472a711b2e9001cbafb2cc289c857b56c732a8e/hal/armv7m/cpu.c#L153 ?

Reply:
@agkaminski , very similarly.

Use:
```
arm-phoenix-gcc -Iinclude/ -Ihal/armv7m -Ihal/armv7m/stm32/ -O2 -Wall -S hal/armv7m/cpu.c
```
to verify, see how helpful gcc assembler output is, gcc will generate the asm output in `cpu.s`, or in the `ia32` case:
```
i386-pc-phoenix-gcc -Iinclude/ -Ihal/ia32/ -O2 -Wall -S hal/ia32/cpu.c
```